### DISCUSSION TITLE
What is the current versioning strategy?

### DISCUSSION THREAD

**Discussion by smyrman** [uptrace/bun#402] [General]
Hi. This is not meant as criticism; I love using Bun, and I will continue to do so, no matter the versioning strategy. But versioning is about communication with the users, so I thought I would ask.

I have noticed that since v1.0.0, all new releases have been releases as PATCH versions (1.0.x). Some of the versions are adding new features, which according to [semantic versioning](https://semver.org/), would trigger MINOR releases.

Now, I don't know how closely this project is aiming to follow semver, and I am not asking you to _change_ your versioning strategy, but as versioning is about communication with the users, I am just curious to know the thoughts behind the current versioning. Which change would trigger a MINOR release, for instance.

Cheers.

**Comment by vmihailenco** [comment]
Hi,

To support meaningful (not necessary semantic) versioning, we need proper development cycles and different branches and whatever else... That requires some extra effort and time which is always in value.

Currently the idea is the following:
- hopefully we don't need major releases with breaking changes
- minor releases happen when there are big new features or big refactoring that can break something subtle
- patch releases for everything else

**Reply by smyrman** [reply]
Thanks for clarifying, that's all I wanted to know.
